src/app/statsGraphs/PieChartGraph.tsx

```diff
@@ -0,0 +1,142 @@
+"use client";
+import React from "react";
+import { PieChart, Pie, Cell, Legend, ResponsiveContainer } from "recharts";
+import { fallbackColor, rankedColors } from "./utils";
+
+type PageSegment = {
+  name: string;
+  value: number;
+  color: string;
+};
+
+interface PieChartGraphProps {
+  data: Record<string, number>;
+}
+
+const PieChartGraph = ({ data }: PieChartGraphProps) => {
+  const sortedEntries = Object.entries(data).sort((a, b) => b[1] - a[1]);
+
+  const categoryData: PageSegment[] = sortedEntries.map(
+    ([name, value], index) => ({
+      name,
+      value,
+      color: rankedColors[index] || fallbackColor,
+    })
+  );
+
+  const renderCustomLabel = (props: any) => {
+    const {
+      cx,
+      cy,
+      midAngle,
+      outerRadius,
+      percent,
+      name,
+    }: {
+      cx: number;
+      cy: number;
+      midAngle?: number;
+      outerRadius: number | string;
+      percent?: number;
+      name: string;
+    } = props;
+
+    if (typeof midAngle !== "number") return null;
+
+    const RADIAN = Math.PI / 180;
+    const radius = Number(outerRadius) + 40;
+    const x = cx + radius * Math.cos(-midAngle * RADIAN);
+    const y = cy + radius * Math.sin(-midAngle * RADIAN);
+
+    const rawName = name.includes("&")
+      ? name.split("&")[0].trim()
+      : name.trim();
+
+    const lines = rawName.includes("-")
+      ? rawName.split("-").map((line) => line.trim())
+      : rawName.length > 10
+      ? rawName.split(" ")
+      : [rawName];
+
+    return (
+      <text
+        x={x}
+        y={y}
+        fill="#000000"
+        textAnchor="middle"
+        dominantBaseline="central"
+        fontSize="1.2rem"
+      >
+        {lines.map((line, index) => (
+          <tspan
+            key={index}
+            x={x}
+            dy={index === 0 ? 0 : "1.2rem"} // Adjust vertical spacing
+          >
+            {line}
+          </tspan>
+        ))}
+      </text>
+    );
+  };
+
+  const renderCustomLegend = () => (
```

Comment:
The same here - I'd extract it out of this context and just pass `categoryData` as input argument.